File: windows/Liferay.SDK/Service/V73/Commercetaxfixedrateaddressrel/CommercetaxfixedrateaddressrelService.cs
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V73.Commercetaxfixedrateaddressrel
{
	public class CommercetaxfixedrateaddressrelService : ServiceBase
	{
		public CommercetaxfixedrateaddressrelService(ISession session)
			: base(session)
		{
		}

		public async Task<dynamic> AddCommerceTaxFixedRateAddressRelAsync(long commerceTaxMethodId, long cpTaxCategoryId, long commerceCountryId, long commerceRegionId, string zip, double rate, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceTaxMethodId", commerceTaxMethodId);
			_parameters.Add("cpTaxCategoryId", cpTaxCategoryId);
			_parameters.Add("commerceCountryId", commerceCountryId);
			_parameters.Add("commerceRegionId", commerceRegionId);
			_parameters.Add("zip", zip);
			_parameters.Add("rate", rate);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxfixedrateaddressrel/add-commerce-tax-fixed-rate-address-rel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> AddCommerceTaxFixedRateAddressRelAsync(long userId, long groupId, long commerceTaxMethodId, long cpTaxCategoryId, long commerceCountryId, long commerceRegionId, string zip, double rate)
		{
			var _parameters = new JsonObject();

			_parameters.Add("userId", userId);
			_parameters.Add("groupId", groupId);
			_parameters.Add("commerceTaxMethodId", commerceTaxMethodId);
			_parameters.Add("cpTaxCategoryId", cpTaxCategoryId);
			_parameters.Add("commerceCountryId", commerceCountryId);
			_parameters.Add("commerceRegionId", commerceRegionId);
			_parameters.Add("zip", zip);
			_parameters.Add("rate", rate);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxfixedrateaddressrel/add-commerce-tax-fixed-rate-address-rel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task DeleteCommerceTaxFixedRateAddressRelAsync(long commerceTaxFixedRateAddressRelId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceTaxFixedRateAddressRelId", commerceTaxFixedRateAddressRelId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxfixedrateaddressrel/delete-commerce-tax-fixed-rate-address-rel", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<dynamic> FetchCommerceTaxFixedRateAddressRelAsync(long commerceTaxFixedRateAddressRelId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceTaxFixedRateAddressRelId", commerceTaxFixedRateAddressRelId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxfixedrateaddressrel/fetch-commerce-tax-fixed-rate-address-rel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<IEnumerable<dynamic>> GetCommerceTaxMethodFixedRateAddressRelsAsync(long groupId, long commerceTaxMethodId, int start, int end, JsonObjectWrapper orderByComparator)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("commerceTaxMethodId", commerceTaxMethodId);
			_parameters.Add("start", start);
			_parameters.Add("end", end);
			this.MangleWrapper(_parameters, "orderByComparator", "$languageUtil.getJSONWrapperClassName($parameter.type)", orderByComparator);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxfixedrateaddressrel/get-commerce-tax-method-fixed-rate-address-rels", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (IEnumerable<dynamic>)_obj;
		}

		public async Task<long> GetCommerceTaxMethodFixedRateAddressRelsCountAsync(long groupId, long commerceTaxMethodId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("commerceTaxMethodId", commerceTaxMethodId);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxfixedrateaddressrel/get-commerce-tax-method-fixed-rate-address-rels-count", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (long)_obj;
		}

		public async Task<dynamic> UpdateCommerceTaxFixedRateAddressRelAsync(long commerceTaxFixedRateAddressRelId, long commerceCountryId, long commerceRegionId, string zip, double rate)
		{
			var _parameters = new JsonObject();

			_parameters.Add("commerceTaxFixedRateAddressRelId", commerceTaxFixedRateAddressRelId);
			_parameters.Add("commerceCountryId", commerceCountryId);
			_parameters.Add("commerceRegionId", commerceRegionId);
			_parameters.Add("zip", zip);
			_parameters.Add("rate", rate);

			var _command = new JsonObject()
			{
				{ "/commerce.commercetaxfixedrateaddressrel/update-commerce-tax-fixed-rate-address-rel", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}
	}
}

File: android/services-v73/src/main/java/com/liferay/mobile/android/v73/commercetaxfixedrateaddressrel/CommercetaxfixedrateaddressrelService.java
/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.mobile.android.v73.commercetaxfixedrateaddressrel;

import com.liferay.mobile.android.service.BaseService;
import com.liferay.mobile.android.service.JSONObjectWrapper;
import com.liferay.mobile.android.service.Session;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

/**
 * @author Bruno Farache
 */
public class CommercetaxfixedrateaddressrelService extends BaseService {

	public CommercetaxfixedrateaddressrelService(Session session) {
		super(session);
	}

	public JSONObject addCommerceTaxFixedRateAddressRel(long userId, long groupId, long commerceTaxMethodId, long cpTaxCategoryId, long commerceCountryId, long commerceRegionId, String zip, double rate) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceCountryId", commerceCountryId);
			_params.put("commerceRegionId", commerceRegionId);
			_params.put("commerceTaxMethodId", commerceTaxMethodId);
			_params.put("cpTaxCategoryId", cpTaxCategoryId);
			_params.put("groupId", groupId);
			_params.put("rate", rate);
			_params.put("userId", userId);
			_params.put("zip", checkNull(zip));

			_command.put("/commerce.commercetaxfixedrateaddressrel/add-commerce-tax-fixed-rate-address-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONObject addCommerceTaxFixedRateAddressRel(long commerceTaxMethodId, long cpTaxCategoryId, long commerceCountryId, long commerceRegionId, String zip, double rate, JSONObjectWrapper serviceContext) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceCountryId", commerceCountryId);
			_params.put("commerceRegionId", commerceRegionId);
			_params.put("commerceTaxMethodId", commerceTaxMethodId);
			_params.put("cpTaxCategoryId", cpTaxCategoryId);
			_params.put("rate", rate);
			_params.put("zip", checkNull(zip));
			mangleWrapper(_params, "serviceContext", "com.liferay.portal.kernel.service.ServiceContext", serviceContext);

			_command.put("/commerce.commercetaxfixedrateaddressrel/add-commerce-tax-fixed-rate-address-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public void deleteCommerceTaxFixedRateAddressRel(long commerceTaxFixedRateAddressRelId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceTaxFixedRateAddressRelId", commerceTaxFixedRateAddressRelId);

			_command.put("/commerce.commercetaxfixedrateaddressrel/delete-commerce-tax-fixed-rate-address-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		session.invoke(_command);
	}

	public JSONObject fetchCommerceTaxFixedRateAddressRel(long commerceTaxFixedRateAddressRelId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceTaxFixedRateAddressRelId", commerceTaxFixedRateAddressRelId);

			_command.put("/commerce.commercetaxfixedrateaddressrel/fetch-commerce-tax-fixed-rate-address-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

	public JSONArray getCommerceTaxMethodFixedRateAddressRels(long groupId, long commerceTaxMethodId, int start, int end, JSONObjectWrapper orderByComparator) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceTaxMethodId", commerceTaxMethodId);
			_params.put("end", end);
			_params.put("groupId", groupId);
			_params.put("start", start);
			mangleWrapper(_params, "orderByComparator", "com.liferay.portal.kernel.util.OrderByComparator<com.liferay.commerce.tax.engine.fixed.model.CommerceTaxFixedRateAddressRel>", orderByComparator);

			_command.put("/commerce.commercetaxfixedrateaddressrel/get-commerce-tax-method-fixed-rate-address-rels", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONArray(0);
	}

	public Integer getCommerceTaxMethodFixedRateAddressRelsCount(long groupId, long commerceTaxMethodId) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceTaxMethodId", commerceTaxMethodId);
			_params.put("groupId", groupId);

			_command.put("/commerce.commercetaxfixedrateaddressrel/get-commerce-tax-method-fixed-rate-address-rels-count", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getInt(0);
	}

	public JSONObject updateCommerceTaxFixedRateAddressRel(long commerceTaxFixedRateAddressRelId, long commerceCountryId, long commerceRegionId, String zip, double rate) throws Exception {
		JSONObject _command = new JSONObject();

		try {
			JSONObject _params = new JSONObject();

			_params.put("commerceCountryId", commerceCountryId);
			_params.put("commerceRegionId", commerceRegionId);
			_params.put("commerceTaxFixedRateAddressRelId", commerceTaxFixedRateAddressRelId);
			_params.put("rate", rate);
			_params.put("zip", checkNull(zip));

			_command.put("/commerce.commercetaxfixedrateaddressrel/update-commerce-tax-fixed-rate-address-rel", _params);
		}
		catch (JSONException jsone) {
			throw new Exception(jsone);
		}

		JSONArray _result = session.invoke(_command);

		if (_result == null) {
			return null;
		}

		return _result.getJSONObject(0);
	}

}